client/state/partner-portal/partner/reducer.ts

```diff
@@ -72,7 +81,12 @@ export const current = withoutPersistence( ( state = initialState.current, actio
 	}
 
 	return state;
-} );
+};
+
+export const current = withSchemaValidation(
+	currentPartnerSchema,
+	withPersistence( currentPartnerReducer )
+);
```

Comment:
We should not be caching the partner and the keys:
1. AFAIK cache goes in localStorage which is NOT secure and should not be used for secrets.
2. If a partner chooses a key (which gets caches), then an a11n disables that key, then the user refreshes the page it will attempt to use the cached key (which no longer exists as it won't be returned by the API) resulting in failing requests and an empty key selection dropdown
![Screenshot 2021-03-24 at 18 13 31](https://user-images.githubusercontent.com/22746396/112344324-a5ee3180-8ccc-11eb-9c90-2f7c4a4cde7c.png)

You need 3 keys to test this as if you have 2 and deactivate one the auto key selection will kick in and hide the bug.

Note: You need to test with 3 CONSECUTIVE active keys as otherwise you will hit this bug: 1187494150150258-as-1200106791469029

---

In summary, we only want to cache the selected key id, not the partner and keys themselves. When the partner and its keys are loaded from the API, we validate the key exists and auto select it.